### DISCUSSION TITLE
Perform variance stabilization transformation (vst) on counts for performing PCA/NMDS

### DISCUSSION THREAD

**Discussion by carloshenriquezc** [ChiLiubio/microeco#384] [Ideas]
There is a way to apply vst on the count matrix to do not rarefy for beta diversity analysis?

**Comment by ChiLiubio** [comment]
How abund performing clr or log (https://chiliubio.github.io/microeco_tutorial/other-examples-1.html#trans_norm)?

**Comment by carloshenriquezc** [comment]
Thanks for the fast reply,,

The trans_norm have several data normalization or transformation
approaches. Please if it is possible, include the vst transformation from
the DEseq2 in the options?

**Comment by ChiLiubio** [comment]
Hi. 'DESeq2' has been implemented in the method parameter. Please see the help document with the command `?trans_norm`

![图片](https://github.com/user-attachments/assets/81fd7f4b-f2df-4a92-bba3-d61539210875)

**Comment by carloshenriquezc** [comment]
Thanks, I'll check it out.

**Comment by carloshenriquezc** [comment]
I tried and receive this warning:

Error in tmp_dataset$add_rownames2taxonomy(use_name = taxa_level) : The
input use_name: OTU has been used in the raw tax_table! Please check it!

**Reply by ChiLiubio** [reply]
Hi. I found it is a minor issue that needed to be fixed. Two solutions are available. The first one is to run this command before doing this: `your_microtable_object$cal_abund()`. The other way is to delete the OTU column from tax_table: `your_microtable_object$tax_table %<>% .[, colnames(.) != "OTU"]`

**Comment by carloshenriquezc** [comment]
I tried the two options, but this doesn't work.

t1 <- trans_norm$new(dataset = dataset)
t1$cal_abund():

Error: attempt to apply non-function
t1$tax_table %<>% .[, colnames(.) != "OTU"]
d1 <- t1$norm(method = "DESeq2",group = "Treatment")
Error in tmp_dataset$add_rownames2taxonomy(use_name = taxa_level) :
  The input use_name: OTU has been used in the raw tax_table! Please check it!

**Reply by ChiLiubio** [reply]
It should be :
```
dataset$tax_table %<>% .[, colnames(.) != "OTU"]
dataset$cal_abund()
t1 <- trans_norm$new(dataset = dataset)
d1 <- t1$norm(method = "DESeq2",group = "Treatment")
```
